### PR TITLE
Retain Continuation.vthread until the J9VMContinuation is freed

### PR DESCRIPTION
**Pull request by babsingh**
This PR is needed to enable the Skynet benchmark.

The changes in eclipse-openj9#18180 pass threadObject to
walkContinuationStackFrames.

The GC can walk the continuations until the associated native
J9VMContinuation structure is freed.

Since we rely on Continuation.vthread in walkContinuationStackFrames,
Continuation.vthread can only be unset once the native continuation
structure is freed.

Now, we set Continuation.vthread to NULL after the continuation is
freed. This will prevent a segfault which can occur due to
eclipse-openj9#18180.

Continuation.vthread is also used in JVMTI to not suspend virtual
threads once they enter the last unmount phase. This won't work anymore
because Continuation.vthread is no longer set to NULL at the start of
the last unmount phase. To address this issue, another continuation
state has been added to indicate that a virtual thread's continuation
has entered the last unmount phase.

Signed-off-by: Babneet Singh <sbabneet@ca.ibm.com>

### CODE REVIEW

File: runtime/jvmti/jvmtiThread.c [hunk 1]
```diff
@@ -1324,8 +1324,11 @@ jvmtiGetCurrentThread(jvmtiEnv *env,
 static jvmtiIterationControl
 jvmtiSuspendResumeCallBack(J9VMThread *vmThread, J9MM_IterateObjectDescriptor *object, void *userData)
 {
-	j9object_t vthread = J9VMJDKINTERNALVMCONTINUATION_VTHREAD(vmThread, object->object);
-	if (NULL != vthread) {
+	j9object_t continuationObj = object->object;
+	j9object_t vthread = J9VMJDKINTERNALVMCONTINUATION_VTHREAD(vmThread, continuationObj);
+	ContinuationState continuationState = J9VMJDKINTERNALVMCONTINUATION_STATE(vmThread, continuationObj);;
+
+	if ((NULL != vthread) && J9_ARE_NO_BITS_SET(continuationState, J9_GC_CONTINUATION_STATE_LAST_UNMOUNT)) {
```
Comment (babsingh) [inline runtime/jvmti/jvmtiThread.c:1331]: @fengxue-IS This source code is in C; we can't invoke C++ code from `ContinuationHelpers.hpp` here; so, I have employed a workaround.

**Review by fengxue-IS — APPROVED** [review]
LGTM

**Comment by babsingh** [comment]
jenkins test sanity zlinux jdk21

**Comment by babsingh** [comment]
jenkins compile win jdk21

**Review by tajila — APPROVED** [review]

**Comment by tajila** [comment]
jenkins test sanity alinux64 jdk21

**Comment by tajila** [comment]
jenkins compile win jdk8

**Pull request merged**